### ISSUE TITLE
Unmount of component not called?

### ISSUE THREAD

**Issue by zaceno** [yelouafi/petit-dom#14]
Hi!

See here: https://codepen.io/zaceno/pen/WJwmge

I'm essentially trying to define a component `<FadeOut>` which ... you guessed it: fades out it's content.

My idea was to make a basic "pass-through" component like this:

```
class FadeOut {
    mount(_, child) {
        this.vnode = child
        return mount(child)
    }
    patch (_, __, ___, child) {
        var oldVNode = this.vnode
        this.vnode = child
        return patch(this.vnode, oldVNode)
    }
    unmount (el) {
        console.log('UNMOUNTING', this)
    }
}

```

...And then (hopefully) I would somehow be able to access the element of the child (perhaps by capturing it earlier in `mount` or `patch`) so I could fade it out slowly before removing it for real.

But apparently `unmount` is not even called! Is this a bug, or am I missing something?

**Comment by yelouafi** [comment]
This is a bug;  the lib should call `unmount` on the old vnode when replacing elements

https://github.com/yelouafi/petit-dom/blob/5412b18ad26108762ebf5624ac1c5bd70c578e57/src/vdom.js#L222-L226

**Comment by zaceno** [comment]
Ok thanks! Glad to know I'm not completely off the trail at least 😄

**Comment by yelouafi** [comment]
should be fixed in [v 0.2.0](https://github.com/yelouafi/petit-dom/releases/tag/v0.2.0)
